monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -105,3 +83,28 @@ def _upsert_from_tcp_scan_event(
         )
 
         self._machine_repository.upsert_network_services(target_machine.id, network_services)
+
+    @classmethod
+    def _get_tcp_connections_from_event(cls, event: TCPScanEvent) -> Sequence[SocketAddress]:
+        unique_connections = {
+            SocketAddress(ip=event.target, port=port) for port in cls._get_open_ports(event)
+        }
+        return tuple(unique_connections)
+
+    @classmethod
+    def _get_new_network_services_from_event(
+        cls, event: TCPScanEvent, target_machine: Machine
+    ) -> NetworkServices:
+        new_services: NetworkServices = {}
+
+        for port in cls._get_open_ports(event):
+            socket_address = SocketAddress(ip=event.target, port=port)
+            if socket_address in target_machine.network_services:
+                # The TCPScanEvent contains no information about services, so this method always
+                # uses NetworkService.Unknown. If a service has already been discovered for this
+                # socket address, then it should not be overwritten.
+                continue
```

Comment:
Do we want to replace the network service if the existing one is UNKNOWN?

Reply:
We'll only have that kind of information (the kind that we can use to update network services) in `FingerprintingEvents`. Skipping a socket address here doesn't matter since it'll always be assigned `NetworkService.UNKNOWN`.

Reply:
I had to think about this for a second too, though, so let's add a comment about it.

Reply:
We discussed on webex and decided to leave this as is.